src/Enum.php

```diff
@@ -178,7 +178,10 @@ public static function __callStatic($name, $arguments)
     {
         $array = static::toArray();
         if (isset($array[$name])) {
-            return new static($array[$name]);
+            $self = new \ReflectionClass(static::class);
+            $arguments = array_merge([$array[$name]], $arguments);
+
+            return $self->newInstanceArgs($arguments);
```

Comment:
With splat operator this should refactored to just `return new static($array[$name], ...$arguments);`